### DISCUSSION TITLE
gitpod access - restricted?

### DISCUSSION THREAD

**Discussion by nisha0119** [opensafely/documentation#311] [Installing OpenSAFELY] [answered]
Has anyone else received an error upon trying to access Gitpod? 
(link embedded in following -> https://docs.opensafely.org/getting-started/)

I seem to be encountering the following error regardless of browser (attempted both Chrome and Firefox):

_We are having trouble connecting to the server.
Either you are offline or websocket connections are blocked_

Investigated further and looks like it could be administrator restrictions.

![Chrome restriction msg](https://user-images.githubusercontent.com/83028350/125423516-ec160097-cbc0-46d7-bc66-d0f0aa1539bb.jpg)
![Firefox restriction msg](https://user-images.githubusercontent.com/83028350/125423524-04f4d1e3-3cf2-4142-8a94-f1e6ee9fab25.jpg)
 
Sharing of any success stories attempting to fix this access issue would be greatly appreciated!

**Comment by StevenMaude** [comment, marked as answer]
@nisha0119: thanks for including screenshots and the error message, they were helpful to see what was going on :+1:

I found a [similar issue](https://github.com/gitpod-io/gitpod/issues/1158) reported for Gitpod. As that issue mentions:

> You'll need to allow websocket connections to *.gitpod.io. We cannot fix that for you, unfortunately.

So, if you are accessing Gitpod via a network that you don't control  — including a VPN when working off-site — then you should request access from your IT/network administrators.

But, the error message you saw seems to be a Gitpod-specific message. So, I think it's more likely that you have some access to Gitpod and just the *websocket* connections required by Gitpod are blocked. Many web services do not use websockets at all: a network administrator might only allow websocket connections for certain sites.

Websockets being blocked is a likely cause of your problem if the Gitpod *homepage* is accessible, but the Gitpod *development environments* are not.

**Reply by nisha0119** [reply]
Thanks @StevenMaude   
Thought it may have been a direct access connection issue (not even sure if that's VPN but thought may be the case). However, it doesn't give me the option to disconnect from that, so I'll give my network admin a call and see if they've got flexibility to unblock the websocket. Wish me luck!

**Reply by sebbacon** [reply]
The alternative (which you have probably already considered!) is to install everything locally, rather than going via Gitpod? It takes a while, but assuming you have institutional permission to install the software, probably worth the hassle to get it all on your own laptop.

**Reply by nisha0119** [reply]
Thanks @sebbacon 
Definitely would go down the route of locally installing if I could but I'm afraid work laptop so facing permission blockers. However, looks like that may also be the case with gitpod as well. Let's see. Will keep you posted on outcome.
